docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -1167,16 +1170,403 @@ payload:
+to store the EdgeDB Auth identity and a new ``current_user`` type.
+
+.. code-block:: sdl-diff
+    :caption: dbschema/default.esdl
+    
+    + global current_user := assert_single(
+    +     ((
+    +         select User { id, name }
+    +         filter .identity = global ext::auth::ClientTokenIdentity
+    +     ))
+    + );
+      
+      type User extending Auditable {
+    +    required identity: ext::auth::Identity;
+         required name: str {
+            constraint exclusive;
+            constraint max_len_value(50);
+         };
+      }
+
+After updating the schema, run the following command to apply the changes:
+
+.. code-block:: bash
+
+    $ edgedb migration create
+    $ edgedb migrate
+
+Next, update the ``create_user_async_edgeql`` query to include the identity:
+
+.. code-block:: edgeql-diff
+    :caption: app/queries/create_user.edgeql
+
+      select (
+    +     with identity := (select ext::auth::Identity filter .id = <uuid>$identity_id),
+          insert User {
+            name := <str>$name,
+    +       identity := identity
+          }) {
+          name,
+          created_at,
+      };
+
+Run ``edgedb-py`` to generate the new function. Now, let's update the sign-up
+endpoint to create a new user in the database. We need to do a few things:
+
+1. Import ``edgedb``.
+
+2. Create an EdgeDB client.
+
+3. Get the identity ID from the EdgeDB Auth server response.
+
+4. Create a new user in the database using the ``create_user_async_edgeql``
+   query.
+
+
+.. code-block:: python-diff
+
+    + import edgedb
+    + client = edgedb.create_async_client()
+
+      @router.post("/auth/signup")
+      async def handle_signup(request: Request):
+          body = await request.json()
+          email = body.get("email")
+          name = body.get("name")
+          password = body.get("password")
+      
+          if not email or not password or not name:
+              raise HTTPException(status_code=400, detail="Missing email, password, or name.")
+      
+          verifier, challenge = generate_pkce()
+          register_url = f"{EDGEDB_AUTH_BASE_URL}/register"
+          response = httpx.post(register_url, json={
+              "challenge": challenge,
+              "email": email,
+              "password": password,
+              "provider": "builtin::local_emailpassword",
+              "verify_url": "http://localhost:8000/auth/verify",
+          })
+      
+          if response.status_code != 200 and response.status_code != 201:
+              return JSONResponse(status_code=400, content={"message": "Registration failed"})
+          
+    +     code = response.json().get("code")
+    +     token_url = f"{EDGEDB_AUTH_BASE_URL}/token"
+    +     token_response = httpx.get(token_url, params={"code": code, "verifier": verifier})
+    + 
+    +     if token_response.status_code != 200:
+    +         return JSONResponse(status_code=400, content={"message": "Token exchange failed"})
```

Comment:
Oh, I see we're doing the code exchange here! We should move it up to the previous endpoint definition and add the token cookie rather than the verifier cookie.